{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "from math import sqrt\n",
    "import numpy as np\n",
    "import sklearn\n",
    "from sklearn.neural_network import MLPRegressor\n",
    "from sklearn.ensemble import RandomForestRegressor\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from numpy import concatenate\n",
    "from matplotlib import pyplot\n",
    "from pandas import read_csv\n",
    "from pandas import DataFrame\n",
    "from pandas import concat\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.metrics import mean_squared_error\n",
    "from keras.models import Sequential\n",
    "from keras.layers import Dense\n",
    "from keras.layers import LSTM\n",
    "from pandas import read_csv\n",
    "from datetime import datetime\n",
    "from keras.layers import Bidirectional\n",
    "import datetime\n",
    "from keras.layers import RepeatVector\n",
    "from keras.layers import TimeDistributed\n",
    "from numpy import array\n",
    "import time\n",
    "import torch\n",
    "import torch.nn as nn\n",
    "import torch.optim as optim\n",
    "from torch.utils.data import Dataset, DataLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def rmse(pred, actual):\n",
    "    error = np.subtract(pred, actual)\n",
    "    sqerror= np.sum(np.square(error))/actual.shape[0]\n",
    "    return np.sqrt(sqerror)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Plot_Mean(name,Overall_Analysis,n_steps_out):\n",
    "    labels = ['TrainRMSE(Mean)','TestRMSE(Mean)']\n",
    "    MLP_Adam=[Overall_Analysis[0][0],Overall_Analysis[0][5]]\n",
    "    MLP_Sgd=[Overall_Analysis[1][0],Overall_Analysis[1][5]]\n",
    "    LSTM=[Overall_Analysis[2][0],Overall_Analysis[2][5]]\n",
    "    Bi_LSTM=[Overall_Analysis[3][0],Overall_Analysis[3][5]]\n",
    "    EN_DC=[Overall_Analysis[4][0],Overall_Analysis[4][5]]\n",
    "    RNN=[Overall_Analysis[5][0],Overall_Analysis[5][5]]\n",
    "    \n",
    "    yer1=np.array([Overall_Analysis[0][3]-Overall_Analysis[0][0],Overall_Analysis[0][8]-Overall_Analysis[0][5]])\n",
    "    yer2=np.array([Overall_Analysis[1][3]-Overall_Analysis[1][0],Overall_Analysis[1][8]-Overall_Analysis[1][5]])\n",
    "    yer3=np.array([Overall_Analysis[2][3]-Overall_Analysis[2][0],Overall_Analysis[2][8]-Overall_Analysis[2][5]])\n",
    "    yer4=np.array([Overall_Analysis[3][3]-Overall_Analysis[3][0],Overall_Analysis[3][8]-Overall_Analysis[3][5]])\n",
    "    yer5=np.array([Overall_Analysis[4][3]-Overall_Analysis[4][0],Overall_Analysis[4][8]-Overall_Analysis[4][5]])\n",
    "    yer6=np.array([Overall_Analysis[5][3]-Overall_Analysis[5][0],Overall_Analysis[5][8]-Overall_Analysis[5][5]])\n",
    "    \n",
    "    width = 0.15  # the width of the bars\n",
    "    Plot(name,labels,width,MLP_Adam,MLP_Sgd,LSTM,Bi_LSTM,EN_DC,RNN,yer1,yer2,yer3,yer4,yer5,yer6,\"\",\"\",\"Train&Test_RMSE_Mean_Comparison\",4)\n",
    "\n",
    "def Plot_Step_RMSE_Mean(name,Overall_Analysis,n_steps_out):\n",
    "    MLP_Adam=Overall_Analysis[0,10:n_steps_out*5+10:5]\n",
    "    MLP_Sgd=Overall_Analysis[1,10:n_steps_out*5+10:5]\n",
    "    LSTM=Overall_Analysis[2,10:n_steps_out*5+10:5]\n",
    "    Bi_LSTM=Overall_Analysis[3,10:n_steps_out*5+10:5]\n",
    "    EN_DC=Overall_Analysis[4,10:n_steps_out*5+10:5]\n",
    "    RNN=Overall_Analysis[5,10:n_steps_out*5+10:5]\n",
    "    yer1=np.subtract(Overall_Analysis[0,13:n_steps_out*5+10:5],MLP_Adam)\n",
    "    yer2=np.subtract(Overall_Analysis[1,13:n_steps_out*5+10:5],MLP_Sgd)\n",
    "    yer3=np.subtract(Overall_Analysis[2,13:n_steps_out*5+10:5],LSTM)\n",
    "    yer4=np.subtract(Overall_Analysis[3,13:n_steps_out*5+10:5],Bi_LSTM)\n",
    "    yer5=np.subtract(Overall_Analysis[4,13:n_steps_out*5+10:5],EN_DC)\n",
    "    yer6=np.subtract(Overall_Analysis[5,13:n_steps_out*5+10:5],RNN)\n",
    "    labels = []\n",
    "    for j in range(n_steps_out):\n",
    "        labels=np.concatenate((labels,[str(j+1)]))\n",
    "    width = 0.13  # the width of the bars\n",
    "    Plot(name,labels,width,MLP_Adam,MLP_Sgd,LSTM,Bi_LSTM,EN_DC,RNN,yer1,yer2,yer3,yer4,yer5,yer6,\"Steps\",\"RMSE(Mean)\",\"Step_RMSE_Comparison\",2)\n",
    "    \n",
    "\n",
    "def Plot(name,labels,width,MLP_Adam,MLP_Sgd,LSTM,Bi_LSTM,EN_DC,RNN,yer1,yer2,yer3,yer4,yer5,yer6,xlabel,ylabel,Gname,cap):\n",
    "    r1 = np.arange(len(labels))\n",
    "    r2 = [x + width for x in r1]\n",
    "    r3 = [x + width for x in r2]\n",
    "    r4 = [x + width for x in r3]\n",
    "    r5 = [x + width for x in r4]\n",
    "    r6 = [x + width for x in r5]\n",
    "\n",
    "    fig, ax = plt.subplots()\n",
    "    rects1 = ax.bar(r1, MLP_Adam, width,edgecolor = 'black', yerr=yer1, capsize=cap, label='FNN-Adam')\n",
    "    rects2 = ax.bar(r2, MLP_Sgd, width,edgecolor = 'black', yerr=yer2,capsize=cap,  label='FNN-SGD')\n",
    "    rects3 = ax.bar(r3, LSTM, width,edgecolor = 'black', yerr=yer3,capsize=cap,  label='LSTM')\n",
    "    rects4 = ax.bar(r4, Bi_LSTM, width,edgecolor = 'black', yerr=yer4,capsize=cap,  label='BD-LSTM')\n",
    "    rects5 = ax.bar(r5, EN_DC, width,edgecolor = 'black', yerr=yer5,capsize=cap,  label='ED-LSTM')\n",
    "    rects6 = ax.bar(r6, RNN, width,edgecolor = 'black', yerr=yer6,capsize=cap,  label='RNN')\n",
    "    \n",
    "    plt.xlabel(xlabel)\n",
    "    plt.ylabel(ylabel)\n",
    "    plt.xticks([r + width for r in range(len(MLP_Adam))], labels)\n",
    "    plt.setp(ax.get_xticklabels(), fontsize=8)\n",
    "    ax.legend()\n",
    "    fig.tight_layout()\n",
    "    plt.savefig(\"Results/\"+name+\"/\"+Gname+\".png\",dpi=300)\n",
    "    plt.show()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def MODEL_FNN_adam(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    mlp_adam = MLPRegressor(hidden_layer_sizes=(Hidden,), activation='relu', solver='adam', alpha=0.1,max_iter=Epochs, tol=0)\n",
    "    Best_RMSE=1000   #Assigning a large number   \n",
    "    \n",
    "    start_time=time.time()\n",
    "    for run in range(Num_Exp):\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        mlp_adam.fit(x_train, y_train)\n",
    "        y_predicttrain = mlp_adam.predict(x_train)\n",
    "        y_predicttest = mlp_adam.predict(x_test)\n",
    "        train_acc[run] = rmse( y_predicttrain, y_train) \n",
    "        print(\"TrainRMSE_Mean\",train_acc[run]) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test)\n",
    "        print(\"TestRMSE_Mean\",test_acc[run])\n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j], y_test[:,j])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test\n",
    "\n",
    "def MODEL_FNN_sgd(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    mlp_sgd = MLPRegressor(hidden_layer_sizes=(Hidden, ), activation='relu', solver='sgd', alpha=0.1,max_iter=Epochs, tol=0)\n",
    "    Best_RMSE=1000   #Assigning a large number  \n",
    "    \n",
    "    start_time=time.time()\n",
    "    for run in range(Num_Exp):\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        mlp_sgd.fit(x_train,y_train)\n",
    "        y_predicttrain = mlp_sgd.predict(x_train)\n",
    "        y_predicttest = mlp_sgd.predict(x_test)\n",
    "        train_acc[run] = rmse( y_predicttrain,y_train) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test) \n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j], y_test[:,j])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test\n",
    " \n",
    "    \n",
    "def MODEL_LSTM(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    n_features = 1\n",
    "    x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], n_features))\n",
    "    print(x_train.shape)\n",
    "    x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], n_features))\n",
    "    print(x_test.shape)\n",
    "    \n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    \n",
    "    model = Sequential()\n",
    "    model.add(LSTM(Hidden, activation='relu', input_shape=(n_steps_in,n_features)))\n",
    "    model.add(Dense(n_steps_out))\n",
    "    model.compile(optimizer='adam', loss='mse')\n",
    "    model.summary()\n",
    "    Best_RMSE=1000   #Assigning a large number \n",
    "    \n",
    "    start_time=time.time()\n",
    "    for run in range(Num_Exp):\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        # fit model\n",
    "        model.fit(x_train, y_train, epochs=Epochs,batch_size=64, verbose=0, shuffle=False)\n",
    "        y_predicttrain = model.predict(x_train)\n",
    "        y_predicttest = model.predict(x_test)\n",
    "        train_acc[run] = rmse( y_predicttrain,y_train) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test) \n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j], y_test[:,j])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test\n",
    "\n",
    "\n",
    "def MODEL_Bi_LSTM(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    n_features = 1\n",
    "    x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], n_features))\n",
    "    print(x_train.shape)\n",
    "    x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], n_features))\n",
    "    print(x_test.shape)\n",
    "    \n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    \n",
    "    model = Sequential()\n",
    "    model.add(Bidirectional(LSTM(Hidden, activation='relu'), input_shape=(n_steps_in,n_features)))\n",
    "    model.add(Dense(n_steps_out))\n",
    "    model.compile(optimizer='adam', loss='mse')\n",
    "    model.summary()\n",
    "    \n",
    "    Best_RMSE=1000   #Assigning a large number \n",
    "    start_time=time.time()\n",
    "    for run in range(Num_Exp):\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        # fit model\n",
    "        model.fit(x_train, y_train, epochs=Epochs,batch_size=64, verbose=0, shuffle=False)\n",
    "        y_predicttrain = model.predict(x_train)\n",
    "        y_predicttest = model.predict(x_test)\n",
    "        train_acc[run] = rmse( y_predicttrain,y_train) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test) \n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j], y_test[:,j])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test\n",
    "\n",
    "\n",
    "def MODEL_EN_DC(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    n_features = 1\n",
    "    x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], n_features))\n",
    "    print(x_train.shape)\n",
    "    x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], n_features))\n",
    "    print(x_test.shape)\n",
    "    y_train = y_train.reshape((y_train.shape[0], y_train.shape[1], n_features))\n",
    "    print(y_train.shape)\n",
    "    y_test = y_test.reshape((y_test.shape[0], y_test.shape[1], n_features))\n",
    "    print(y_test.shape)\n",
    "    \n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    \n",
    "    model = Sequential()\n",
    "    model.add(LSTM(Hidden, activation='relu',input_shape=(n_steps_in,n_features)))\n",
    "    model.add(RepeatVector(n_steps_out))\n",
    "    model.add(LSTM(Hidden, activation='relu', return_sequences=True))\n",
    "    model.add(TimeDistributed(Dense(1,activation='relu')))\n",
    "    model.compile(optimizer='adam', loss='mse')\n",
    "    model.summary()\n",
    "    \n",
    "    Best_RMSE=1000   #Assigning a large number \n",
    "    start_time=time.time()\n",
    "    for run in range(Num_Exp):\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        # fit model\n",
    "        model.fit(x_train, y_train, epochs=Epochs,batch_size=64, verbose=0, shuffle=False)\n",
    "        y_predicttrain = model.predict(x_train)\n",
    "        y_predicttest = model.predict(x_test)\n",
    "        train_acc[run] = rmse( y_predicttrain,y_train) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test) \n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j,0], y_test[:,j,0])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test.reshape(y_test.shape[0], y_test.shape[1])\n",
    "\n",
    "class RNN_Model(nn.Module):\n",
    "    def __init__(self, batch_size, n_inputs, n_neurons, n_outputs):\n",
    "        super(RNN_Model, self).__init__()\n",
    "        self.batch_size=batch_size\n",
    "        self.n_inputs=n_inputs\n",
    "        self.n_neurons=n_neurons\n",
    "        self.n_outputs=n_outputs\n",
    "        self.rnn = nn.RNNCell(n_inputs, n_neurons)\n",
    "        self.hx = torch.zeros(batch_size, n_neurons) # initialize hidden state\n",
    "        self.FC = nn.Linear(n_neurons, n_outputs)\n",
    "    def forward(self, X):\n",
    "        self.hx = torch.zeros(self.batch_size, self.n_neurons)\n",
    "        self.hx = self.rnn(X, self.hx)\n",
    "        out = self.FC(self.hx)  \n",
    "        out=out.reshape(out.size(1))\n",
    "        return out\n",
    "\n",
    "def MODEL_RNN(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden):\n",
    "    train_acc=np.zeros(Num_Exp)\n",
    "    test_acc=np.zeros(Num_Exp)\n",
    "    Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "    Best_RMSE=1000   #Assigning a large number \n",
    "    start_time=time.time()\n",
    "      \n",
    "    # Define hyperparameters\n",
    "    n_epochs = Epochs\n",
    "    lr=0.01\n",
    "    \n",
    "    x_train = torch.tensor(x_train, dtype = torch.float)\n",
    "    x_test = torch.tensor(x_test, dtype = torch.float)\n",
    "    y_train = torch.tensor(y_train, dtype = torch.float)\n",
    "    x_train=x_train.resize_((x_train.size(0),1,x_train.size(1)))\n",
    "    x_test=x_test.resize_((x_test.size(0),1,x_test.size(1)))\n",
    "    for run in range(Num_Exp):\n",
    "        model = RNN_Model(1, n_steps_in,Hidden,n_steps_out)\n",
    "        # Define Loss, Optimizer\n",
    "        criterion = nn.MSELoss()\n",
    "        optimizer = torch.optim.SGD(model.parameters(), lr=lr)\n",
    "        print(\"Experiment\",run+1,\"in progress\")\n",
    "        for epoch in range(0, n_epochs):\n",
    "            for i in range(x_train.size(0)):\n",
    "                optimizer.zero_grad()    # Clears existing gradients from previous epoch\n",
    "                output_val = model(x_train[i])\n",
    "                loss = criterion(output_val, y_train[i])\n",
    "                loss.backward(retain_graph=True)     # Does backpropagation and calculates gradients\n",
    "                optimizer.step()     # Updates the weights accordingly \n",
    "        \n",
    "        y_predicttrain=[]\n",
    "        for i in range(x_train.size(0)):\n",
    "            output_val = model(x_train[i])\n",
    "            y_predicttrain.append(output_val.tolist())\n",
    "        y_predicttrain=np.array(y_predicttrain)\n",
    "        \n",
    "        y_predicttest=[] \n",
    "        for i in range(x_test.size(0)):\n",
    "            output_val = model(x_test[i])\n",
    "            y_predicttest.append(output_val.tolist())\n",
    "        y_predicttest=np.array(y_predicttest)\n",
    "        \n",
    "        train_acc[run] = rmse( y_predicttrain,np.array(y_train.tolist()))\n",
    "        print(\"TrainRMSE_Mean\",train_acc[run]) \n",
    "        test_acc[run] = rmse( y_predicttest, y_test) \n",
    "        print(\"TestRMSE_Mean\",test_acc[run]) \n",
    "        \n",
    "        if test_acc[run]<Best_RMSE:\n",
    "            Best_RMSE=test_acc[run]\n",
    "            Best_Predict_Test=y_predicttest\n",
    "        for j in range(n_steps_out):\n",
    "            Step_RMSE[run][j]=rmse(y_predicttest[:,j], y_test[:,j])\n",
    "            \n",
    "    print(\"Total time for\",Num_Exp,\"experiments\",time.time()-start_time)\n",
    "    return train_acc,test_acc,Step_RMSE,Best_Predict_Test.reshape(y_test.shape[0], y_test.shape[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lazer\n",
      "Experiment 1 in progress\n",
      "TrainRMSE_Mean 0.3311659131452987\n",
      "TestRMSE_Mean 0.24727983711184365\n",
      "Experiment 2 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3302909311312988\n",
      "TestRMSE_Mean 0.25131595179289495\n",
      "Experiment 3 in progress\n",
      "TrainRMSE_Mean 0.33410136154243353\n",
      "TestRMSE_Mean 0.26665861386135137\n",
      "Experiment 4 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3418599949094983\n",
      "TestRMSE_Mean 0.255586317655468\n",
      "Experiment 5 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.35476857190936767\n",
      "TestRMSE_Mean 0.2528109208881873\n",
      "Experiment 6 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3672408581380609\n",
      "TestRMSE_Mean 0.2733014948734987\n",
      "Experiment 7 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3288765608067143\n",
      "TestRMSE_Mean 0.24210084964282275\n",
      "Experiment 8 in progress\n",
      "TrainRMSE_Mean 0.32903065340843496\n",
      "TestRMSE_Mean 0.25735013403293944\n",
      "Experiment 9 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3406230025172732\n",
      "TestRMSE_Mean 0.26158748240003354\n",
      "Experiment 10 in progress\n",
      "TrainRMSE_Mean 0.332385189591783\n",
      "TestRMSE_Mean 0.2467392877862791\n",
      "Experiment 11 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3468387165639003\n",
      "TestRMSE_Mean 0.2621062453331975\n",
      "Experiment 12 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.33347027176062194\n",
      "TestRMSE_Mean 0.2525247598793551\n",
      "Experiment 13 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3374383758700555\n",
      "TestRMSE_Mean 0.24858927850754062\n",
      "Experiment 14 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3438107365231724\n",
      "TestRMSE_Mean 0.26183071978756783\n",
      "Experiment 15 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.342840514918535\n",
      "TestRMSE_Mean 0.2648133447564361\n",
      "Experiment 16 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.32847171401280767\n",
      "TestRMSE_Mean 0.25139990893102676\n",
      "Experiment 17 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.34731402581062976\n",
      "TestRMSE_Mean 0.24722459853634315\n",
      "Experiment 18 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.36644042254624537\n",
      "TestRMSE_Mean 0.27123673146278426\n",
      "Experiment 19 in progress\n",
      "TrainRMSE_Mean 0.3380712128406405\n",
      "TestRMSE_Mean 0.2580219012638322\n",
      "Experiment 20 in progress\n",
      "TrainRMSE_Mean 0.33932359705577864\n",
      "TestRMSE_Mean 0.25498174863309714\n",
      "Experiment 21 in progress\n",
      "TrainRMSE_Mean 0.33305762609092676\n",
      "TestRMSE_Mean 0.25320247332378987\n",
      "Experiment 22 in progress\n",
      "TrainRMSE_Mean 0.33902277642010537\n",
      "TestRMSE_Mean 0.2515707462159002\n",
      "Experiment 23 in progress\n",
      "TrainRMSE_Mean 0.6009644132213834\n",
      "TestRMSE_Mean 0.4033776256765858\n",
      "Experiment 24 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n",
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n",
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3638147457668266\n",
      "TestRMSE_Mean 0.2648272963263671\n",
      "Experiment 25 in progress\n",
      "TrainRMSE_Mean 0.5022906664071568\n",
      "TestRMSE_Mean 0.3467990511743882\n",
      "Experiment 26 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.4663464977407008\n",
      "TestRMSE_Mean 0.32997421389985143\n",
      "Experiment 27 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.3947638400391951\n",
      "TestRMSE_Mean 0.28541533212048004\n",
      "Experiment 28 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.34234322169107145\n",
      "TestRMSE_Mean 0.25906002783376375\n",
      "Experiment 29 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (1000) reached and the optimization hasn't converged yet.\n",
      "  % self.max_iter, ConvergenceWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.32997375559092296\n",
      "TestRMSE_Mean 0.24364229066891818\n",
      "Experiment 30 in progress\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/sklearn/neural_network/_multilayer_perceptron.py:573: UserWarning: Training interrupted by user.\n",
      "  warnings.warn(\"Training interrupted by user.\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrainRMSE_Mean 0.34539353054927396\n",
      "TestRMSE_Mean 0.25634122042376595\n",
      "Total time for 30 experiments 24.150845050811768\n",
      "MODEL_FNN_adam\n",
      "       TrainRMSE  TestRMSE    Step1    Step2    Step3    Step4    Step5  \\\n",
      "Exp1     0.33117   0.24728  0.06937  0.10169  0.07690  0.07502  0.07173   \n",
      "Exp2     0.33029   0.25132  0.06534  0.10416  0.07961  0.08038  0.07439   \n",
      "Exp3     0.33410   0.26666  0.08091  0.12350  0.07333  0.08046  0.06676   \n",
      "Exp4     0.34186   0.25559  0.07706  0.10343  0.07742  0.07258  0.07813   \n",
      "Exp5     0.35477   0.25281  0.07391  0.09803  0.07765  0.07254  0.08773   \n",
      "Exp6     0.36724   0.27330  0.08895  0.11051  0.07711  0.08571  0.07464   \n",
      "Exp7     0.32888   0.24210  0.07992  0.10427  0.07033  0.07781  0.07169   \n",
      "Exp8     0.32903   0.25735  0.07428  0.10693  0.08536  0.08503  0.07394   \n",
      "Exp9     0.34062   0.26159  0.06839  0.11013  0.08414  0.08383  0.08032   \n",
      "Exp10    0.33239   0.24674  0.07628  0.10022  0.07001  0.07477  0.07733   \n",
      "Exp11    0.34684   0.26211  0.08010  0.10667  0.07796  0.07705  0.08616   \n",
      "Exp12    0.33347   0.25252  0.07589  0.11916  0.06832  0.07420  0.07997   \n",
      "Exp13    0.33744   0.24859  0.06905  0.10004  0.08147  0.07624  0.07276   \n",
      "Exp14    0.34381   0.26183  0.08925  0.10106  0.08180  0.06941  0.08501   \n",
      "Exp15    0.34284   0.26481  0.07300  0.11872  0.08557  0.07651  0.08405   \n",
      "Exp16    0.32847   0.25140  0.06373  0.10673  0.08311  0.08061  0.07403   \n",
      "Exp17    0.34731   0.24722  0.06519  0.10406  0.07626  0.08188  0.06615   \n",
      "Exp18    0.36644   0.27124  0.08024  0.11087  0.08185  0.07455  0.09588   \n",
      "Exp19    0.33807   0.25802  0.08481  0.09566  0.07480  0.08216  0.07942   \n",
      "Exp20    0.33932   0.25498  0.07227  0.09849  0.07945  0.06997  0.07971   \n",
      "Exp21    0.33306   0.25320  0.07322  0.10951  0.08069  0.07950  0.07281   \n",
      "Exp22    0.33902   0.25157  0.07919  0.10540  0.06960  0.07762  0.07706   \n",
      "Exp23    0.60096   0.40338  0.14254  0.16575  0.09768  0.09478  0.08278   \n",
      "Exp24    0.36381   0.26483  0.08861  0.10665  0.06978  0.08132  0.07753   \n",
      "Exp25    0.50229   0.34680  0.10375  0.10416  0.09153  0.08576  0.11021   \n",
      "Exp26    0.46635   0.32997  0.08023  0.10852  0.09627  0.06831  0.10323   \n",
      "Exp27    0.39476   0.28542  0.09179  0.10460  0.08361  0.09219  0.09085   \n",
      "Exp28    0.34234   0.25906  0.07589  0.10686  0.08375  0.07698  0.07431   \n",
      "Exp29    0.32997   0.24364  0.07569  0.10007  0.07325  0.07311  0.07220   \n",
      "Exp30    0.34539   0.25634  0.07114  0.10754  0.07623  0.08074  0.07540   \n",
      "\n",
      "         Step6    Step7    Step8    Step9   Step10  \n",
      "Exp1   0.07035  0.06273  0.07073  0.08309  0.09228  \n",
      "Exp2   0.07442  0.06137  0.06154  0.08955  0.09272  \n",
      "Exp3   0.06889  0.06114  0.06250  0.10421  0.09874  \n",
      "Exp4   0.06859  0.06493  0.07212  0.09297  0.09249  \n",
      "Exp5   0.07490  0.05857  0.06517  0.08290  0.09845  \n",
      "Exp6   0.08202  0.06733  0.07375  0.09866  0.09628  \n",
      "Exp7   0.06475  0.05792  0.05695  0.08591  0.08393  \n",
      "Exp8   0.06578  0.06226  0.06840  0.08866  0.09247  \n",
      "Exp9   0.07165  0.06298  0.06387  0.09330  0.09582  \n",
      "Exp10  0.06775  0.06440  0.06054  0.08781  0.09183  \n",
      "Exp11  0.07199  0.06772  0.07552  0.08864  0.09022  \n",
      "Exp12  0.06704  0.06004  0.05588  0.08980  0.08920  \n",
      "Exp13  0.07195  0.05940  0.06745  0.08259  0.09593  \n",
      "Exp14  0.07187  0.06945  0.06770  0.09188  0.09288  \n",
      "Exp15  0.06749  0.07082  0.06013  0.09310  0.09278  \n",
      "Exp16  0.06616  0.06803  0.06208  0.08684  0.09199  \n",
      "Exp17  0.07135  0.06219  0.06026  0.08345  0.09794  \n",
      "Exp18  0.07940  0.06318  0.06689  0.09665  0.09631  \n",
      "Exp19  0.07320  0.06389  0.07336  0.09169  0.09138  \n",
      "Exp20  0.07589  0.07166  0.07235  0.08978  0.09133  \n",
      "Exp21  0.06916  0.06287  0.05857  0.08828  0.09309  \n",
      "Exp22  0.07116  0.06646  0.06799  0.08823  0.08493  \n",
      "Exp23  0.13012  0.14619  0.11701  0.14783  0.12529  \n",
      "Exp24  0.07577  0.06418  0.07528  0.09456  0.09457  \n",
      "Exp25  0.11337  0.17672  0.10430  0.09868  0.07795  \n",
      "Exp26  0.10015  0.06026  0.18417  0.09028  0.10173  \n",
      "Exp27  0.09457  0.07765  0.06871  0.09821  0.09493  \n",
      "Exp28  0.06800  0.05653  0.06911  0.09248  0.10161  \n",
      "Exp29  0.07002  0.06203  0.05998  0.08342  0.09158  \n",
      "Exp30  0.06436  0.06134  0.07106  0.08937  0.10064  \n",
      "              Mean  Standard Deviation    CI_LB    CI_UB      Min      Max\n",
      "TrainRMSE  0.36108             0.05868  0.34008  0.38208  0.32847  0.60096\n",
      "TestRMSE   0.26739             0.03370  0.25533  0.27945  0.24210  0.40338\n",
      "Step1      0.07967             0.01451  0.07447  0.08486  0.06373  0.14254\n",
      "Step2      0.10811             0.01233  0.10370  0.11252  0.09566  0.16575\n",
      "Step3      0.07950             0.00718  0.07692  0.08207  0.06832  0.09768\n",
      "Step4      0.07870             0.00609  0.07652  0.08088  0.06831  0.09478\n",
      "Step5      0.07987             0.00975  0.07639  0.08336  0.06615  0.11021\n",
      "Step6      0.07607             0.01460  0.07085  0.08130  0.06436  0.13012\n",
      "Step7      0.07047             0.02502  0.06152  0.07943  0.05653  0.17672\n",
      "Step8      0.07311             0.02408  0.06450  0.08173  0.05588  0.18417\n",
      "Step9      0.09243             0.01151  0.08831  0.09655  0.08259  0.14783\n",
      "Step10     0.09438             0.00760  0.09166  0.09710  0.07795  0.12529\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KeyboardInterrupt\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 3331, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-5-0b8d0550fc48>\", line 192, in <module>\n",
      "    if __name__ == \"__main__\": main()\n",
      "  File \"<ipython-input-5-0b8d0550fc48>\", line 160, in main\n",
      "    plt.savefig(\"Results/\"+name+\"/\"+Mname+'/pred_Step'+str(j+1)+'.png',dpi=300)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/pyplot.py\", line 729, in savefig\n",
      "    res = fig.savefig(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/figure.py\", line 2180, in savefig\n",
      "    self.canvas.print_figure(fname, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backend_bases.py\", line 2091, in print_figure\n",
      "    **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py\", line 532, in print_png\n",
      "    self.figure.dpi, metadata=metadata)\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'SystemError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1148, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 742, in getmodule\n",
      "    os.path.realpath(f)] = module.__name__\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 395, in realpath\n",
      "    path, ok = _joinrealpath(filename[:0], filename, {})\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 429, in _joinrealpath\n",
      "    if not islink(newpath):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 171, in islink\n",
      "    st = os.lstat(path)\n",
      "KeyboardInterrupt\n",
      "KeyboardInterrupt\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 3331, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-5-0b8d0550fc48>\", line 192, in <module>\n",
      "    if __name__ == \"__main__\": main()\n",
      "  File \"<ipython-input-5-0b8d0550fc48>\", line 160, in main\n",
      "    plt.savefig(\"Results/\"+name+\"/\"+Mname+'/pred_Step'+str(j+1)+'.png',dpi=300)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/pyplot.py\", line 729, in savefig\n",
      "    res = fig.savefig(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/figure.py\", line 2180, in savefig\n",
      "    self.canvas.print_figure(fname, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backend_bases.py\", line 2091, in print_figure\n",
      "    **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py\", line 532, in print_png\n",
      "    self.figure.dpi, metadata=metadata)\n",
      "SystemError: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'SystemError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 3254, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 3348, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1415, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1315, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1183, in structured_traceback\n",
      "    formatted_exceptions += self.prepare_chained_exception_message(evalue.__cause__)\n",
      "TypeError: can only concatenate str (not \"list\") to str\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1148, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 742, in getmodule\n",
      "    os.path.realpath(f)] = module.__name__\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 395, in realpath\n",
      "    path, ok = _joinrealpath(filename[:0], filename, {})\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 429, in _joinrealpath\n",
      "    if not islink(newpath):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/posixpath.py\", line 171, in islink\n",
      "    st = os.lstat(path)\n",
      "KeyboardInterrupt\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"list\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: ",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.FileIO object at 0x12228aac30> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mSystemError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;31mSystemError\u001b[0m: <built-in method write of _io.BufferedWriter object at 0x111ee5d10> returned a result with an error set",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mshowtraceback\u001b[0;34m(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)\u001b[0m\n\u001b[1;32m   2043\u001b[0m                         \u001b[0;31m# in the engines. This should return a list of strings.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2044\u001b[0;31m                         \u001b[0mstb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_render_traceback_\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2045\u001b[0m                     \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'SystemError' object has no attribute '_render_traceback_'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_code\u001b[0;34m(self, code_obj, result, async_)\u001b[0m\n\u001b[1;32m   3346\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3347\u001b[0m                 \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merror_in_exec\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3348\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshowtraceback\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrunning_compiled_code\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3349\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3350\u001b[0m             \u001b[0moutflag\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mshowtraceback\u001b[0;34m(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)\u001b[0m\n\u001b[1;32m   2045\u001b[0m                     \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2046\u001b[0m                         stb = self.InteractiveTB.structured_traceback(etype,\n\u001b[0;32m-> 2047\u001b[0;31m                                             value, tb, tb_offset=tb_offset)\n\u001b[0m\u001b[1;32m   2048\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2049\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_showtraceback\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0metype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[0;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[1;32m   1413\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtb\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1414\u001b[0m         return FormattedTB.structured_traceback(\n\u001b[0;32m-> 1415\u001b[0;31m             self, etype, value, tb, tb_offset, number_of_lines_of_context)\n\u001b[0m\u001b[1;32m   1416\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1417\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[0;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[1;32m   1313\u001b[0m             \u001b[0;31m# Verbose modes need a full traceback\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1314\u001b[0m             return VerboseTB.structured_traceback(\n\u001b[0;32m-> 1315\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtb_offset\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnumber_of_lines_of_context\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1316\u001b[0m             )\n\u001b[1;32m   1317\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mmode\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'Minimal'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[0;34m(self, etype, evalue, etb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[1;32m   1181\u001b[0m         \u001b[0mexception\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_parts_of_chained_exception\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mevalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1182\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1183\u001b[0;31m             \u001b[0mformatted_exceptions\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_chained_exception_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mevalue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__cause__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1184\u001b[0m             \u001b[0metype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mevalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1185\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: can only concatenate str (not \"list\") to str"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error in callback <function flush_figures at 0x1a26100710> (for post_execute):\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/events.py\", line 88, in trigger\n",
      "    func(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/ipykernel/pylab/backend_inline.py\", line 121, in flush_figures\n",
      "    return show(True)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/ipykernel/pylab/backend_inline.py\", line 43, in show\n",
      "    metadata=_fetch_figure_metadata(figure_manager.canvas.figure)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/display.py\", line 313, in display\n",
      "    format_dict, md_dict = format(obj, include=include, exclude=exclude)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/formatters.py\", line 180, in format\n",
      "    data = formatter(obj)\n",
      "  File \"<decorator-gen-9>\", line 2, in __call__\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/formatters.py\", line 224, in catch_format_error\n",
      "    r = method(self, *args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/formatters.py\", line 341, in __call__\n",
      "    return printer(obj)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/pylabtools.py\", line 248, in <lambda>\n",
      "    png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/pylabtools.py\", line 132, in print_figure\n",
      "    fig.canvas.print_figure(bytes_io, **kw)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backend_bases.py\", line 2091, in print_figure\n",
      "    **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py\", line 527, in print_png\n",
      "    FigureCanvasAgg.draw(self)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py\", line 388, in draw\n",
      "    self.figure.draw(self.renderer)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/artist.py\", line 38, in draw_wrapper\n",
      "    return draw(artist, renderer, *args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/figure.py\", line 1709, in draw\n",
      "    renderer, self, artists, self.suppressComposite)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/image.py\", line 135, in _draw_list_compositing_images\n",
      "    a.draw(renderer)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/artist.py\", line 38, in draw_wrapper\n",
      "    return draw(artist, renderer, *args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/axes/_base.py\", line 2647, in draw\n",
      "    mimage._draw_list_compositing_images(renderer, self, artists)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/image.py\", line 135, in _draw_list_compositing_images\n",
      "    a.draw(renderer)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/artist.py\", line 38, in draw_wrapper\n",
      "    return draw(artist, renderer, *args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/axis.py\", line 1205, in draw\n",
      "    renderer)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/axis.py\", line 1150, in _get_tick_bboxes\n",
      "    for tick in ticks if tick.label1.get_visible()],\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/axis.py\", line 1150, in <listcomp>\n",
      "    for tick in ticks if tick.label1.get_visible()],\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/text.py\", line 890, in get_window_extent\n",
      "    bbox, info, descent = self._get_layout(self._renderer)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/text.py\", line 338, in _get_layout\n",
      "    M = Affine2D().rotate_deg(self.get_rotation())\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/transforms.py\", line 1938, in rotate_deg\n",
      "    return self.rotate(np.deg2rad(degrees))\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/matplotlib/transforms.py\", line 1926, in rotate\n",
      "    self._mtx = np.dot(rotate_mtx, self._mtx)\n",
      "  File \"<__array_function__ internals>\", line 6, in dot\n",
      "KeyboardInterrupt\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'KeyboardInterrupt' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 1148, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/site-packages/IPython/core/ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 739, in getmodule\n",
      "    f = getabsfile(module)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 708, in getabsfile\n",
      "    _filename = getsourcefile(object) or getfile(object)\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/inspect.py\", line 693, in getsourcefile\n",
      "    if os.path.exists(filename):\n",
      "  File \"/Users/shauryagoyal32/opt/anaconda3/envs/tensorflow/lib/python3.7/genericpath.py\", line 19, in exists\n",
      "    os.stat(path)\n",
      "KeyboardInterrupt\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m"
     ]
    }
   ],
   "source": [
    "def main():\n",
    "    \n",
    "    n_steps_in, n_steps_out = 5,10\n",
    "    \n",
    "#     Overall_Analysis=np.zeros([42,10+n_steps_out*5])\n",
    "    Overall_Analysis = pd.read_csv(\"Results/OverallAnalysis.csv\",index_col = 0)\n",
    "    Overall_Analysis=Overall_Analysis.values\n",
    "    Overall_Analysis=Overall_Analysis[:,2:]\n",
    "    for i in range(1,8):\n",
    "        problem=i\n",
    "        if problem ==1:\n",
    "            TrainData = pd.read_csv(\"../data/Lazer/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Lazer/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name= \"Lazer\"\n",
    "        if problem ==2:\n",
    "            TrainData = pd.read_csv(\"../data/Sunspot/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Sunspot/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name= \"Sunspot\"\n",
    "        if problem ==3:\n",
    "            TrainData = pd.read_csv(\"../data/Mackey/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Mackey/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name=\"Mackey\"\n",
    "        if problem ==4:\n",
    "            TrainData = pd.read_csv(\"../data/Lorenz/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Lorenz/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values  \n",
    "            name= \"Lorenz\"\n",
    "        if problem ==5:\n",
    "            TrainData = pd.read_csv(\"../data/Rossler/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Rossler/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name= \"Rossler\"\n",
    "        if problem ==6:\n",
    "            TrainData = pd.read_csv(\"../data/Henon/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/Henon/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name= \"Henon\"\n",
    "        if problem ==7:\n",
    "            TrainData = pd.read_csv(\"../data/ACFinance/train1.csv\",index_col = 0)\n",
    "            TrainData = TrainData.values\n",
    "            TestData = pd.read_csv(\"../data/ACFinance/test1.csv\",index_col = 0)\n",
    "            TestData = TestData.values\n",
    "            name= \"ACFinance\" \n",
    "\n",
    "        x_train = TrainData[:,0:n_steps_in]\n",
    "        y_train = TrainData[:,n_steps_in : n_steps_in+n_steps_out ]\n",
    "        x_test = TestData[:,0:n_steps_in]\n",
    "        y_test = TestData[:,n_steps_in : n_steps_in+n_steps_out]\n",
    "\n",
    "        print(name)\n",
    "        Num_Exp=30    #No. of experiments\n",
    "        Epochs=1000\n",
    "        Hidden=10\n",
    "        train_acc=np.zeros(Num_Exp)\n",
    "        test_acc=np.zeros(Num_Exp)\n",
    "        Step_RMSE=np.zeros([Num_Exp,n_steps_out])\n",
    "\n",
    "\n",
    "        for k in range(1,7):\n",
    "\n",
    "            method=k\n",
    "            if method ==1:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_FNN_adam(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_FNN_adam\"\n",
    "            if method ==2:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_FNN_sgd(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_FNN_sgd\"\n",
    "            if method ==3:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_LSTM(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_LSTM\"\n",
    "            if method ==4:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_Bi_LSTM(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_Bi_LSTM\"\n",
    "            if method ==5:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_EN_DC(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_EN_DC\"\n",
    "            if method ==6:\n",
    "                train_acc,test_acc,Step_RMSE,Best_Predict_Test=MODEL_RNN(x_train,x_test,y_train,y_test,Num_Exp,n_steps_in,n_steps_out,Epochs,Hidden)\n",
    "                Mname=\"MODEL_RNN\"\n",
    "\n",
    "            print(Mname)\n",
    "\n",
    "            arr = np.dstack((train_acc,test_acc))\n",
    "            arr=arr.reshape(Num_Exp,2)\n",
    "            arr=np.concatenate((arr,Step_RMSE), axis=1)\n",
    "            arr=arr.reshape(Num_Exp,2+n_steps_out)\n",
    "            \n",
    "            ExpIndex=np.array([])\n",
    "            for j in range(Num_Exp):\n",
    "                ExpIndex=np.concatenate((ExpIndex,[\"Exp\"+str(j+1)]))\n",
    "\n",
    "            ExpIndex1=['TrainRMSE','TestRMSE']\n",
    "            for j in range(n_steps_out):\n",
    "                ExpIndex1=np.concatenate((ExpIndex1,[\"Step\"+str(j+1)]))\n",
    "                \n",
    "            arr=np.round_(arr, decimals = 5) \n",
    "            arr = pd.DataFrame(arr, index = ExpIndex , columns = ExpIndex1)\n",
    "            arr.to_csv(\"Results/\"+name+\"/\"+Mname+\"/ExpAnalysis.csv\")\n",
    "            print(arr)\n",
    "            \n",
    "            Train_Mean=np.mean(train_acc)\n",
    "            Train_Std=np.std(train_acc)\n",
    "            Train_CI_LB= Train_Mean-1.96*(Train_Std/np.sqrt(Num_Exp))\n",
    "            Train_CI_UB= Train_Mean+1.96*(Train_Std/np.sqrt(Num_Exp))\n",
    "            \n",
    "            Test_Mean=np.mean(test_acc)\n",
    "            Test_Std=np.std(test_acc)\n",
    "            Test_CI_LB= Test_Mean-1.96*(Test_Std/np.sqrt(Num_Exp))\n",
    "            Test_CI_UB= Test_Mean+1.96*(Test_Std/np.sqrt(Num_Exp))\n",
    "            \n",
    "            Overall_Analysis[(i-1)*6+(k-1)][0]=Train_Mean\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][1]=Train_Std\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][2]=Train_CI_LB\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][3]=Train_CI_UB\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][4]=np.min(train_acc)\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][5]=Test_Mean\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][6]=Test_Std\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][7]=Test_CI_LB\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][8]=Test_CI_UB\n",
    "            Overall_Analysis[(i-1)*6+(k-1)][9]=np.min(test_acc)\n",
    "            \n",
    "            arr1 = np.vstack(([Train_Mean,Train_Std,Train_CI_LB,Train_CI_UB,np.min(train_acc),np.max(train_acc)],[Test_Mean,Test_Std,Test_CI_LB,Test_CI_UB,np.min(test_acc),np.max(test_acc)]))\n",
    "            \n",
    "            for j in range(n_steps_out):\n",
    "                Step_mean = np.mean(Step_RMSE[:,j])\n",
    "                Step_std = np.std(Step_RMSE[:,j])\n",
    "                Step_min = np.min(Step_RMSE[:,j])\n",
    "                Step_CI_LB= Step_mean-1.96*(Step_std/np.sqrt(Num_Exp))\n",
    "                Step_CI_UB= Step_mean+1.96*(Step_std/np.sqrt(Num_Exp))\n",
    "                arr1=np.vstack((arr1,[Step_mean,Step_std,Step_CI_LB,Step_CI_UB,Step_min,np.max(Step_RMSE[:,j])]))\n",
    "                Overall_Analysis[(i-1)*6+(k-1)][5*j+10]= Step_mean\n",
    "                Overall_Analysis[(i-1)*6+(k-1)][5*j+11]= Step_std\n",
    "                Overall_Analysis[(i-1)*6+(k-1)][5*j+12]= Step_CI_LB\n",
    "                Overall_Analysis[(i-1)*6+(k-1)][5*j+13]= Step_CI_UB\n",
    "                Overall_Analysis[(i-1)*6+(k-1)][5*j+14]= Step_min\n",
    "            arr1=np.round_(arr1, decimals = 5) \n",
    "            arr1 = pd.DataFrame(arr1, index=ExpIndex1, columns = ['Mean','Standard Deviation','CI_LB','CI_UB','Min','Max'])\n",
    "            print(arr1)\n",
    "            arr1.to_csv(\"Results/\"+name+\"/\"+Mname+\"/OverallAnalysis.csv\")\n",
    "            \n",
    "            \n",
    "            x_data=np.linspace(0,y_test.shape[0], num=y_test.shape[0])\n",
    "            for j in range(n_steps_out):\n",
    "                plt.figure()\n",
    "                plt.plot(x_data, y_test[:,j], label='actual')\n",
    "                plt.plot(x_data, Best_Predict_Test[:,j], label='predicted')\n",
    "                plt.ylabel('RMSE')  \n",
    "                plt.xlabel('Time (samples)') \n",
    "                plt.title('Actual vs Predicted')\n",
    "                plt.legend()\n",
    "                plt.savefig(\"Results/\"+name+\"/\"+Mname+'/pred_Step'+str(j+1)+'.png',dpi=300) \n",
    "                plt.show()\n",
    "                plt.close()\n",
    "            \n",
    "        \n",
    "        \n",
    "        #Plot mean of train_RMSE and test_RMSE\n",
    "        #Plot Std of train_RMSE and test_RMSE\n",
    "        Plot_Mean_Std(name,Overall_Analysis[6*(i-1):(6*i),:],n_steps_out)\n",
    "        #Plot Step wise RMSE mean for different methods\n",
    "        Plot_Step_RMSE_Mean(name,Overall_Analysis[6*(i-1):(6*i),:],n_steps_out)\n",
    "    \n",
    "    Overall_Analysis=Overall_Analysis.astype('float64')\n",
    "    Overall_Analysis=np.round_(Overall_Analysis, decimals = 5)  \n",
    "    Index1=[]\n",
    "    for j in range(7):\n",
    "        Index1=np.concatenate((Index1, ['FNN-Adam','FNN-SGD','LSTM','BD-LSTM','ED-LSTM','RNN']))\n",
    "    Index2=[\"Lazer\",\"Sunspot\",\"Mackey\",\"Lorenz\",\"Rossler\",\"Henon\",\"ACFinance\"]\n",
    "    Index2=np.repeat(Index2,6)\n",
    "    Index=np.dstack((Index2,Index1))\n",
    "    Index=Index.reshape(42,2)\n",
    "    Column=['Dataset','Method','Train-RMSE_Mean','Train-RMSE_Std','Train-CI-LB','Train-CI-UB','TrainRMSE_Min','TestRMSE-Mean','TestRMSE-Std','Test-CI-LB','Test-CI-UB','Test-RMSE-Min']\n",
    "    for j in range(1,11):\n",
    "        Column=np.concatenate((Column, ['Step'+str(j)+'-RMSE-Mean','Step'+str(j)+'-RMSE-Std','Step'+str(j)+'-CI-LB','Step'+str(j)+'-CI-UB','Step'+str(j)+'-RMSE-Min']))\n",
    "  \n",
    "    Overall_Analysis=np.concatenate((Index,Overall_Analysis), axis=1)\n",
    "    Overall_Analysis = pd.DataFrame(Overall_Analysis, columns = Column)\n",
    "    print(Overall_Analysis)\n",
    "    Overall_Analysis.to_csv(\"Results/OverallAnalysis.csv\")\n",
    "    \n",
    "    \n",
    "    \n",
    "if __name__ == \"__main__\": main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.33708        0.25375         0.07463         0.10426   \n",
      "FNN-SGD          0.42512        0.28205         0.08951          0.0983   \n",
      "LSTM             0.19539        0.19105         0.05772         0.07246   \n",
      "BD-LSTM          0.16191        0.20067         0.04391         0.08917   \n",
      "ED-LSTM          0.11659        0.20195           0.049         0.08943   \n",
      "RNN              0.32105        0.25803         0.06412         0.11757   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.08198         0.07638         0.07605         0.06906   \n",
      "FNN-SGD          0.08155         0.08524         0.08743         0.07874   \n",
      "LSTM             0.08067         0.06969          0.0512         0.05399   \n",
      "BD-LSTM          0.07733         0.05471         0.05962         0.06554   \n",
      "ED-LSTM          0.07072         0.06011         0.06935         0.06056   \n",
      "RNN              0.08323         0.07617         0.07552         0.07304   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.06319         0.06423         0.08906           0.0924  \n",
      "FNN-SGD          0.07401         0.08637         0.10319          0.09681  \n",
      "LSTM              0.0537         0.04641          0.0507          0.05611  \n",
      "BD-LSTM          0.06011         0.04597         0.05992          0.06307  \n",
      "ED-LSTM          0.05817         0.05105         0.05268          0.06146  \n",
      "RNN              0.06426         0.06106         0.08822          0.09471  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.20427        0.15104          0.0163         0.02356   \n",
      "FNN-SGD          0.32295        0.21787         0.02807         0.03521   \n",
      "LSTM              0.1418        0.11603         0.00719         0.01482   \n",
      "BD-LSTM          0.13694        0.11472         0.00857         0.01549   \n",
      "ED-LSTM          0.12095        0.13224         0.01091         0.01699   \n",
      "RNN              0.18748        0.13423          0.0132         0.02119   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.03108         0.03495         0.04068         0.04645   \n",
      "FNN-SGD          0.04411         0.05185         0.06098         0.06773   \n",
      "LSTM             0.02197         0.02751         0.03213         0.03597   \n",
      "BD-LSTM          0.02218         0.02763         0.03178         0.03583   \n",
      "ED-LSTM          0.02366         0.02921         0.03477         0.04021   \n",
      "RNN              0.02847         0.03459         0.03946         0.04311   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam          0.0514         0.05816         0.06534          0.07449  \n",
      "FNN-SGD          0.07713         0.08558         0.09312          0.10121  \n",
      "LSTM              0.0397         0.04489         0.05086           0.0587  \n",
      "BD-LSTM           0.0395         0.04399         0.04983          0.05756  \n",
      "ED-LSTM          0.04582         0.05186         0.05905          0.06733  \n",
      "RNN               0.0463         0.05029         0.05644          0.06411  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.18099        0.18218          0.0162         0.02559   \n",
      "FNN-SGD           0.2576        0.25986         0.04852         0.06208   \n",
      "LSTM             0.08903         0.0897         0.00564         0.00801   \n",
      "BD-LSTM          0.07499        0.07649         0.00516         0.00832   \n",
      "ED-LSTM          0.05867        0.06018          0.0059         0.00764   \n",
      "RNN              0.13172         0.1321         0.00776         0.01419   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.03985         0.04565         0.05205         0.05808   \n",
      "FNN-SGD          0.06862          0.0745         0.07851         0.08796   \n",
      "LSTM             0.01205         0.01783         0.02384         0.02976   \n",
      "BD-LSTM          0.01166         0.01553         0.02023         0.02441   \n",
      "ED-LSTM          0.01026         0.01329         0.01675         0.02001   \n",
      "RNN              0.02139         0.02895         0.03649         0.04342   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.06804         0.07265         0.07608          0.07767  \n",
      "FNN-SGD          0.09122         0.09368         0.09632          0.09903  \n",
      "LSTM             0.03413         0.03807         0.04058          0.04185  \n",
      "BD-LSTM          0.02883         0.03179         0.03432          0.03586  \n",
      "ED-LSTM          0.02273         0.02475         0.02609          0.02713  \n",
      "RNN              0.04962          0.0547         0.05864           0.0615  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.19318        0.18087         0.01827         0.02059   \n",
      "FNN-SGD          0.27609        0.26494         0.03363         0.04321   \n",
      "LSTM              0.0242        0.02535         0.00246         0.00326   \n",
      "BD-LSTM          0.03001        0.03104         0.00431         0.00545   \n",
      "ED-LSTM          0.02252        0.02342         0.00511         0.00438   \n",
      "RNN              0.05381        0.05417         0.01129         0.01286   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.02533         0.03335         0.04811         0.05268   \n",
      "FNN-SGD          0.05468         0.06506         0.07872         0.08658   \n",
      "LSTM             0.00416         0.00513         0.00642         0.00731   \n",
      "BD-LSTM          0.00644         0.00742         0.00789         0.00942   \n",
      "ED-LSTM          0.00465         0.00518         0.00593         0.00682   \n",
      "RNN              0.01432         0.01511         0.01554         0.01643   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.06132         0.06782         0.08846          0.08586  \n",
      "FNN-SGD          0.09441         0.10269         0.11155          0.11777  \n",
      "LSTM             0.00892         0.01008         0.01167          0.01293  \n",
      "BD-LSTM          0.01068         0.01251         0.01326          0.01461  \n",
      "ED-LSTM           0.0079         0.00898         0.00995          0.01095  \n",
      "RNN              0.01709         0.01864         0.01989          0.02265  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.15461        0.14727         0.01484         0.02017   \n",
      "FNN-SGD           0.3757        0.43144         0.04673          0.0666   \n",
      "LSTM             0.04158        0.04875           0.008         0.00862   \n",
      "BD-LSTM          0.02814        0.03486         0.00381         0.00468   \n",
      "ED-LSTM          0.03741        0.04268         0.00846         0.00817   \n",
      "RNN              0.10877        0.10305         0.01858         0.02177   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.02517         0.03225         0.03995         0.04896   \n",
      "FNN-SGD          0.09102         0.10597         0.12573         0.14239   \n",
      "LSTM             0.00988         0.01166         0.01351         0.01553   \n",
      "BD-LSTM          0.00606         0.00724         0.00844         0.01005   \n",
      "ED-LSTM          0.00976         0.01121         0.01284         0.01408   \n",
      "RNN              0.02501         0.02901         0.03136         0.03394   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.05269         0.06027         0.06212          0.06734  \n",
      "FNN-SGD          0.15725         0.16641         0.18179          0.18813  \n",
      "LSTM             0.01701         0.01848         0.02036          0.02246  \n",
      "BD-LSTM          0.01203         0.01415         0.01567          0.01779  \n",
      "ED-LSTM          0.01511         0.01589         0.01671          0.01803  \n",
      "RNN              0.03605         0.03817         0.04036           0.0424  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.54695        0.53784         0.14652         0.16063   \n",
      "FNN-SGD          0.56701        0.55775         0.17253         0.17113   \n",
      "LSTM             0.45424        0.45164          0.0287         0.06825   \n",
      "BD-LSTM          0.40144        0.41271          0.0241         0.04478   \n",
      "ED-LSTM          0.32346        0.32938         0.02258         0.04536   \n",
      "RNN              0.52473        0.51619         0.08851         0.15146   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam         0.16103         0.17135         0.17312         0.17582   \n",
      "FNN-SGD          0.17069         0.17597         0.17694         0.17772   \n",
      "LSTM             0.09198         0.13862         0.15844         0.16417   \n",
      "BD-LSTM          0.06095         0.09246         0.12867         0.15379   \n",
      "ED-LSTM          0.05168          0.0609          0.0694         0.08677   \n",
      "RNN              0.15771         0.16431         0.17178         0.17303   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.17856         0.17806         0.17569          0.17625  \n",
      "FNN-SGD           0.1814          0.1805         0.17875          0.17734  \n",
      "LSTM              0.1684         0.17073         0.17227          0.17565  \n",
      "BD-LSTM          0.15929         0.16974         0.17375          0.17328  \n",
      "ED-LSTM          0.11203         0.13713         0.15241          0.16887  \n",
      "RNN              0.17636         0.17683         0.17519          0.17506  \n",
      "         Train-RMSE-Mean Test-RMSE-Mean Step1-RMSE-Mean Step2-RMSE-Mean  \\\n",
      "FNN-Adam         0.16284        0.08849         0.01647         0.02028   \n",
      "FNN-SGD          0.17033         0.0988         0.02092         0.02417   \n",
      "LSTM             0.14707          0.086          0.0127         0.01677   \n",
      "BD-LSTM          0.14543         0.0915         0.01266         0.01648   \n",
      "ED-LSTM          0.14369        0.09226         0.01304         0.01708   \n",
      "RNN              0.19297        0.09365         0.01731          0.0202   \n",
      "\n",
      "         Step3-RMSE-Mean Step4-RMSE-Mean Step5-RMSE-Mean Step6-RMSE-Mean  \\\n",
      "FNN-Adam          0.0217         0.02492         0.02724          0.0289   \n",
      "FNN-SGD           0.0266         0.02774         0.02989         0.03249   \n",
      "LSTM               0.019         0.02201         0.02477         0.02811   \n",
      "BD-LSTM          0.01935         0.02288         0.02533         0.02922   \n",
      "ED-LSTM          0.02038         0.02386          0.0271         0.03023   \n",
      "RNN              0.02283         0.02577         0.02841         0.03041   \n",
      "\n",
      "         Step7-RMSE-Mean Step8-RMSE-Mean Step9-RMSE-Mean Step10-RMSE-Mean  \n",
      "FNN-Adam         0.03113         0.03229         0.03389          0.03575  \n",
      "FNN-SGD          0.03422         0.03498         0.03568          0.03802  \n",
      "LSTM             0.03021         0.03332         0.03639          0.03674  \n",
      "BD-LSTM          0.03309         0.03562         0.03875           0.0409  \n",
      "ED-LSTM          0.03339         0.03593         0.03797          0.03953  \n",
      "RNN              0.03273         0.03485         0.03711          0.03844  \n"
     ]
    }
   ],
   "source": [
    "Overall_Analysis = pd.read_csv(\"Results_28th/Results/OverallAnalysis.csv\",index_col = 0)\n",
    "Overall_Analysis=Overall_Analysis[['Dataset','Method','Train-RMSE-Mean','Test-RMSE-Mean','Step1-RMSE-Mean','Step2-RMSE-Mean','Step3-RMSE-Mean','Step4-RMSE-Mean','Step5-RMSE-Mean','Step6-RMSE-Mean','Step7-RMSE-Mean','Step8-RMSE-Mean','Step9-RMSE-Mean','Step10-RMSE-Mean']]\n",
    "Overall_Analysis=Overall_Analysis.values\n",
    "n_steps_out=10\n",
    "for i in range(1,8):\n",
    "    name=Overall_Analysis[6*(i-1)][0]\n",
    "    temp=Overall_Analysis[6*(i-1):(6*i),2:]\n",
    "    temp = pd.DataFrame(temp, index=['FNN-Adam','FNN-SGD','LSTM','BD-LSTM','ED-LSTM','RNN'], columns = ['Train-RMSE-Mean','Test-RMSE-Mean','Step1-RMSE-Mean','Step2-RMSE-Mean','Step3-RMSE-Mean','Step4-RMSE-Mean','Step5-RMSE-Mean','Step6-RMSE-Mean','Step7-RMSE-Mean','Step8-RMSE-Mean','Step9-RMSE-Mean','Step10-RMSE-Mean'])\n",
    "    print(temp)\n",
    "    temp.to_csv(\"Results_28th/Results/\"+name+\"/Analysis.csv\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
